src/filters/oidc/jwks_storage.h

```diff
@@ -0,0 +1,75 @@
+#ifndef AUTHSERVICE_SRC_FILTERS_OIDC_JWKS_STORAGE_H_
+#define AUTHSERVICE_SRC_FILTERS_OIDC_JWKS_STORAGE_H_
+
+#include <spdlog/spdlog.h>
+
+#include <memory>
+#include <mutex>
+
+#include "jwt_verify_lib/jwks.h"
+#include "src/common/http/http.h"
+
+namespace authservice {
+namespace filters {
+namespace oidc {
+
+class JwksFetcher;
+
+class JwksStorage {
+ public:
+  JwksStorage(const std::string& jwks_uri, boost::asio::io_context& ioc);
+  JwksStorage(const std::string& jwks, google::jwt_verify::Jwks::Type type) {
+    updateJwks(jwks, type);
+  }
+
+  const google::jwt_verify::JwksPtr& jwks() const {
+    std::unique_lock<std::mutex> lck(mux_);
+    return jwks_;
+  }
+
+  void updateJwks(const std::string& new_jwks,
```

Comment:
will this only be invoked once instead of periodic refresh?

Reply:
No. In refactored implementation, `NonPermanentJwksStorage` wiil update existing JWKs but `PermanentJwksStorage` won't do it.